### PR TITLE
svgbobrus: init at 2017-08-29

### PR DESCRIPTION
**Pull request by matthiasbeyer**
###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Pinging @kevincox as requested.

This fails with

```
ERROR: The Cargo.lock file doesn't exist
```

### CODE REVIEW

**Comment by Mic92** [comment]
The solution is to add a `Cargo.lock`. Either ask the project to do so or run `cargo update` in the project and add the Cargo.lock as a patch to the project. Otherwise we can not build this project reproducible.

**Comment by matthiasbeyer** [comment]
And how to refer to the Cargo.lock file then?

**Comment by matthiasbeyer** [comment]
Ah, I see my problem: The commandline binary "svgbob_cli" is actually a subdir of the repository. And there is also the Cargo.lock file.

How to specify a subdir as the actual target?

**Comment by Mic92** [comment]
Set the `sourceRoot` variable would be my wild guess.

**Comment by kevincox** [comment]
Seems like it should work. This is what https://github.com/NixOS/nixpkgs/blob/7c696a08ce0faa7851dec48d9899980699f224a3/pkgs/applications/version-management/pijul/default.nix#L14 does.

**Comment by matthiasbeyer** [comment]
Hm... does not work for me. Also not without the "rus" part.

**Comment by kevincox** [comment]
I'll take a look when I get home but often what I do it add `set -i` somewhere so that I can see what it is doing, that often helps me find out what I am setting wrong. You can see other rust packages to find spots where you can inject this.

File: pkgs/applications/misc/svgbobrus/default.nix [hunk 1]
```diff
@@ -2,20 +2,22 @@
...
     repo = "svgbobrus";
-    rev = "ab1c44b49657e856ab7e03e6c80c6972206504da";
+    inherit rev;
     sha256 = "1bbh6inrif2y8159j4y1nvnrcf3fscrhb3zrshfms4847pms42cs";
   };
 
   depsSha256 = "19lrj4i6vzmf22r6xg99zcwvzjpiar8pqin1m2nvv78xzxx5yvgb";
 
-  sourceRoot = "${name}rus/svgbob_cli";
+  sourceRoot = "svgbobrus-${rev}-src/svgbob_cli";
```
Comment (Mic92) [inline pkgs/applications/misc/svgbobrus/default.nix]: You also need to update `rustRegistry`:

```
Using rust registry from /nix/store/z01ibnk837dchls2xirjm171k2sb51f6-rustRegistry-2017-08-27
warning: custom registry support via the `registry.index` configuration is being removed, this functionality will not work in the future
    Updating registry `file:///nix/store/z01ibnk837dchls2xirjm171k2sb51f6-rustRegistry-2017-08-27`
    Updating registry `file:///nix/store/z01ibnk837dchls2xirjm171k2sb51f6-rustRegistry-2017-08-27`
error: no matching package named `svgbob` found (required by `svgbob_cli`)
location searched: registry file:///nix/store/z01ibnk837dchls2xirjm171k2sb51f6-rustRegistry-2017-08-27
version required: = 0.2.1
versions found: 0.2.0, 0.1.14, 0.1.13, ...
note: keeping build directory ‘/tmp/nix-build-svgbob-fetch.drv-5’
builder for ‘/nix/store/3c5gh2c48k56112apv3wz407dv98iq4i-svgbob-fetch.drv’ failed with exit code 101
cannot build derivation ‘/nix/store/8svy5vvhxr03a8ragj5apyjgyb2a5xyc-svgbob.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/8svy5vvhxr03a8ragj5apyjgyb2a5xyc-svgbob.drv’ failed
/run/current-system/sw/bin/nix-shell: failed to build all dependencies
```

**Comment by matthiasbeyer** [comment]
Okay, now I only get a compiletime error (when compiling svgbobrus) because the nightly toolchain is required instead of stable. Everything else seems to work.

**Comment by kevincox** [comment]
Hmm, I don't actually know how to use nightly. There is a passthru.rustc which might work but I haven't tried it.

**Comment by Mic92** [comment]
There is `rustBeta` and `rustNightly`. However both are broken at the moment in master. `rustNightlyBin.buildRustPackage` might work.

**Comment by orivej** [comment]
#30088 has deleted `rustNightly`. @matthiasbeyer would you try to resurrect or close this PR?

**Comment by matthiasbeyer** [comment]
Well, I do not care about that software this much anymore... shall we close it then?

**Comment by Mic92** [comment]
this is up to you.

**Pull request closed without merging**